### PR TITLE
nixos/less configure less with module

### PR DESCRIPTION
**Pull request by JohnAZoidberg**
###### Motivation for this change
Because less is configured not directly with a plaintext file but the config file requires translation to a binary file I thought this is the perfect use case for a nixos module.
Not only can less be configured declaratively now, but I added the convenience that less by default now extracts (xz, gzip, bz2) compressed files transparently before reading.

I think everything possible in the real configuration can be done with the nix language now - therefore there is no need for an option like "extraConfig".

###### Things done

Added a module to configure less with. Extracts compressed files by default before opening them.

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: nixos/modules/programs/less.nix [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+      (mapAttrsToList (variable: values: "${variable}=${values}") cfg.envVariables)
+    }
+  '';
+
+  lessKey = pkgs.runCommand "lesskey"
+            { src = pkgs.writeText "lessconfig" configFile; }
+            "${pkgs.less}/bin/lesskey -o $out $src";
+
+  lessPipe = pkgs.writeScriptBin "lesspipe.sh" ''
+    #! /bin/sh
```
Comment (JohnAZoidberg) [inline nixos/modules/programs/less.nix]: Should I change it to `#!${stdenv.shell}`?

Comment (Mic92) [reply nixos/modules/programs/less.nix]: yes

Comment (Mic92) [reply nixos/modules/programs/less.nix]: I wonder if [lesspipe](https://github.com/wofr06/lesspipe) would be the better default here. We already package it in nixpkgs.

Comment (JohnAZoidberg) [reply nixos/modules/programs/less.nix]: Yup, that's a good idea. It's (almost) always better to use something well maintained than to hack a script together yourself.

**Comment by Mic92** [comment]
Thanks. Module looks good to me with good examples.

**Pull request merged**